### PR TITLE
docs: Update releases md to 3-10-3

### PR DESCRIPTION
**Pull request by Oleksandr123234**
## Description
This change update the RELEASES.md file to contain the release notes for 3.10.3.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking change which improves an existing feature or documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist:
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Pull Request contains one commit. I squash my commits.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -60,6 +61,68 @@ Get acquainted with the latest KubeRocketCI releases.
 * [Version 2.6.0](#2.6.0)
 </details>
 
+## Version 3.10.3 <a name="3.10.3"></a> (December 12, 2024)
+
+### Upgrades
+
+* KubeRocketCI has been verified on the [0.7.2](https://github.com/projectcapsule/capsule/releases) version of Capsule. ([#88](https://github.com/epam/edp-cd-pipeline-operator/issues/88))
+* KubeRocketCI has been verified on the [1.29](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.30.md#v1300) cluster version. ([#66](https://github.com/epam/edp-cd-pipeline-operator/issues/66))
```
Comment (MykolaMarusenko) [inline RELEASES.md:69]: 1.30

File: RELEASES.md [hunk 2]
```diff
@@ -60,6 +61,68 @@ Get acquainted with the latest KubeRocketCI releases.
...
+* The ability to show pipeline run logs from long-term storages, such as [OpenSearch](https://opensearch.org/) has been added. It can be used when platform couldn't to retrieve these logs from Tekton. ([#505](https://github.com/epam/edp-headlamp/issues/505))
+* The ability to rerun deploy pipelines with variables has been added. ([#461](https://github.com/epam/edp-headlamp/issues/461))
+* The ability to specify a set pipelines and tasks has been added using the [deployableResources](https://github.com/epam/edp-tekton/blob/master/charts/pipelines-library/values.yaml#L14) section of the values.yaml file. ([#346](https://github.com/epam/edp-install/issues/346))
+
+### Enhancements
+
+* KubeRocketCI portal now shows user avatar in the profile icon. ([#509](https://github.com/epam/edp-headlamp/issues/509))
+* To prevent unintended deletions and potential impacts on production environments, deploy pipelines now validate the environment namespace uniqueness. ([#91](https://github.com/epam/edp-cd-pipeline-operator/issues/91))
```
Comment (MykolaMarusenko) [inline RELEASES.md:81]: lets do not mention production env here

File: RELEASES.md [hunk 3]
```diff
@@ -60,6 +61,68 @@ Get acquainted with the latest KubeRocketCI releases.
...
+* A resource quota widget was added to the platform. It displays platform usage and limits. This widget is accessible if the [Capsule](https://docs.kuberocketci.io/docs/next/operator-guide/advanced-installation/capsule) resource quotas are configured. ([#503](https://github.com/epam/edp-headlamp/issues/503))
+* Quality gates of **Manual** type no longer require specifying names. ([#496](https://github.com/epam/edp-headlamp/issues/496))
```
Comment (MykolaMarusenko) [inline RELEASES.md:83]: ?

File: RELEASES.md [hunk 4]
```diff
@@ -60,6 +61,68 @@ Get acquainted with the latest KubeRocketCI releases.
...
+
+### Upgrades
+
+* KubeRocketCI has been verified on the [0.7.2](https://github.com/projectcapsule/capsule/releases) version of Capsule. ([#88](https://github.com/epam/edp-cd-pipeline-operator/issues/88))
+* KubeRocketCI has been verified on the [1.29](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.30.md#v1300) cluster version. ([#66](https://github.com/epam/edp-cd-pipeline-operator/issues/66))
+
+### New Functionality
+
+* A new compact deployment flow view mode has been added to the platform. To enable compact view mode, navigate to deployment flows details page and click **View less details** on the top right corner of the page. ([#511](https://github.com/epam/edp-headlamp/issues/511))
+* The ability to show pipeline run logs from long-term storages, such as [OpenSearch](https://opensearch.org/) has been added. It can be used when platform couldn't to retrieve these logs from Tekton. ([#505](https://github.com/epam/edp-headlamp/issues/505))
```
Comment (Iryna-Churkina) [inline RELEASES.md:74]: rm article

**Review by SergK — APPROVED** [review]

**Pull request merged**